src/DelegateDecompiler/Processor.cs

```diff
@@ -1028,7 +1036,17 @@ static void StElem(ProcessorState state)
             {
                 var elementType = array.Type.GetElementType();
                 var expressions = CreateArrayInitExpressions(elementType, newArray, value, index);
-                array.Expression = Expression.NewArrayInit(elementType, expressions);
+                if (typeof(Expression).IsAssignableFrom(array.Type.GetElementType()))
```

Comment:
changes I'm not confortable with but still work well in use-case UnquoteLinqMethodCallExpression .Lambda